esx_service/utils/kvESX.py

```diff
@@ -42,24 +47,78 @@
 # Default kv side car alignment
 KV_ALIGN = 4096
 
+global using_64bits
+
 # Load the disk lib API library
-def loadDiskLib():
+def loadDiskLib(lib_name):
     global lib
 
     if not lib:
-        lib = CDLL(DISK_LIB)
-        lib.DiskLib_Init.argtypes = []
-        lib.DiskLib_Init.restype = c_bool
-        lib.DiskLib_Init()
+       lib = CDLL(lib_name)
+       lib.DiskLib_Init.argtypes = []
+       lib.DiskLib_Init.restype = c_bool
+       lib.DiskLib_Init()
 
 
 # Loads the ESX library used to access proprietary ESX disk lib API.
 # Create arg/result definitions for those that we use.
-def kvESXInit():
+
+# Initialize 32-bit disk lib interfaces
+def kv_ESX_lib64_init():
+    global useSideCarCreate
+
+    # Load disklib APIs
+    loadDiskLib(DISK_LIB64)
+
+    # Define all of the functions we are interested in
+    lib.DiskLib_OpenWithInfo.argtypes = [c_char_p, c_int32, POINTER(c_uint32),
+                                         POINTER(c_uint64), POINTER(c_uint64)]
+    lib.DiskLib_OpenWithInfo.restype = int
+
+    lib.DiskLib_Close.argtypes = [c_uint64]
+    lib.DiskLib_Close.restype = int
+
+    # Check if this library supports create API
+    try:
+        lib.DiskLib_SidecarCreate.argtypes = [c_uint64, c_char_p, c_uint64,
+                                              c_int32, POINTER(c_uint64)]
+        lib.DiskLib_SidecarCreate.restype = int
+        useSideCarCreate = True
+    except:
+        # do nothing
+        logging.debug(
+            "ESX version doesn't support create API, using open instead.")
+
+    lib.DiskLib_SidecarOpen.argtypes = [c_uint64, c_char_p, c_int32,
+                                        POINTER(c_uint64)]
+    lib.DiskLib_SidecarOpen.restype = int
+
+    lib.DiskLib_SidecarClose.argtypes = [c_uint64, c_char_p, POINTER(c_uint64)]
+    lib.DiskLib_SidecarClose.restype = int
+
+    lib.DiskLib_SidecarMakeFileName.argtypes = [c_char_p, c_char_p]
+    lib.DiskLib_SidecarMakeFileName.restype = c_char_p
+
+    lib.ObjLib_Pread.argtypes = [c_uint32, c_void_p, c_uint64, c_uint64]
```

Comment:
let's add only what we need at the moment. We are not providing generic wrapper, we just use what is needed